src/charm.py

```diff
@@ -127,8 +137,19 @@ def _configure(self, _):
             container.restart(self._name)
             logger.info("Prometheus (re)started")
 
+        # Provide connection info to any remote write clients.
+        self._set_remote_write()
+
         self.unit.status = ActiveStatus()
 
+    def _set_remote_write(self):
+        """Provide connection info to any remote write clients."""
+        if self.model.relations["ingress"] and self.app.planned_units() == 1:
+            self.remote_write.set_endpoint(address=self._external_hostname, port=self._port)
+        else:
+            address = f"{self.unit.name.replace('/','-')}.{self.app.name}-endpoints.{self.model.name}.svc.cluster.local"
+            self.remote_write.set_endpoint(address=address, port=self._port)
```

Comment:
I suspect it is impossible to differentiate ingress from cluster.local this way:
A prometheus unit may have an ingress relation for outsiders to use remote write, and at the same time there may be local charms that would like to use remote write.

Reply:
It is my understanding that if an ingress exists, it should be routable from within the cluster. If not then this needs to get a lot more complicated including config variables in the consumer.

Reply:
This is not universally true, but this risk is one I accept